client/db.go

```diff
@@ -215,6 +215,15 @@ type Store interface {
 	// GetAllIndexes returns all the indexes that currently exist within this [Store].
 	GetAllIndexes(context.Context) (map[CollectionName][]IndexDescription, error)
 
+	// CreateDocIndex creates an index for the given document.
+	CreateDocIndex(context.Context, Collection, *Document) error
```

Comment:
suggestion: Are you sure you want 3 public functions for this?  Is there not only ever one valid index state for a document at any given time, and would this not be more safely handled by a `RefreshIndex(doc)`-like function, that decides for the caller what underlying action(s) should be taken?

Reply:
it's not possible. Indexes operate on values, on ids. To update an indexed document you need to provide a doc with the old value to delete from index and a doc with a new value to add.
Proposed `RefreshIndex` receives only one doc.

Reply:
Ahhh, true - that's a bit annoying.  I wonder if we can get around that somehow by using something clever like tracking the txn or using events, but that can be looked at later, not in the last couple of days before release :)

Thanks for the explanation :)